packages/custom-storage/__tests__/SQLStorage.spec.ts

```diff
@@ -0,0 +1,179 @@
+import { SQL } from '@carto/toolkit-sql';
+import { SQLStorage } from '../src/sql/SQLStorage';
+import {
+  StoredVisualization,
+  // Visualization,
+  // Dataset
+} from '../src/StorageRepository';
+import {
+  CHECK_CREATE_QUERIES,
+  LIST_VISUALIZATION_QUERY
+} from './mocks/MockedQueries';
+
+describe('SQLStorage', () => {
+  let sqlStorage: SQLStorage;
+
+  beforeAll(() => {
+    const sqlClient = new SQL('user', 'm14p1k3y', 'https://{user}.carto.com/');
+    sqlStorage = new SQLStorage('mynamespace', sqlClient, 1, true);
+  });
+
+  it('should check or create tables for storage', async () => {
+    // We use only one mock because is not important for the rest of the queries
+    (global as any).fetch.mockResponse(
+      JSON.stringify({
```

Comment:
What do we use this response for in the test?

Aren't we more interested in the queries sent themselves than in the response?
I was wondering if spying the SQL calls would have been enough, instead of replacing fetch

Reply:
What do you think @jesusbotella ?

Reply:
This response is used for granting access to the tables (the last 3 queries). If this data is not added on response, the method will fail with a `Cannot grant: got no current_user data` exception.

Reply:
Thx, now I see it more clearly!; so the rolename determines the SQL sentences being generated. Perhaps having them inline, using that same rolename (as a shared variable?) could be more clear.